StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetFlowControllerViewController.swift

```diff
@@ -432,6 +432,9 @@ class PaymentSheetFlowControllerViewController: UIViewController, FlowController
 
     @objc
     private func didTapContinueButton() {
+        // The user is continuing with an LPM, so we un-select Link
+        isHackyLinkButtonSelected = false
```

Comment:
We reset the Link selection if the user continues with an LPM. Just dismissing the sheet will no longer set `isHackyLinkButtonSelected` to false, meaning that its selection will persist. This is in line with how LPMs work today.